test/traits/detail_screen_test.dart

```diff
@@ -0,0 +1,99 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:provider/provider.dart';
+import 'package:yonomi_device_widgets/assets/traits/unknown_item_icon.dart';
+import 'package:yonomi_device_widgets/providers/lock_provider.dart';
+import 'package:yonomi_device_widgets/providers/trait_based_device_notifier.dart';
+import 'package:yonomi_device_widgets/traits/detail_screen.dart';
+import 'package:yonomi_device_widgets/traits/lock.dart';
+import 'package:yonomi_platform_sdk/third_party/yonomi_graphql_schema/schema.docs.schema.gql.dart';
+import 'package:yonomi_platform_sdk/yonomi-sdk.dart';
+
+import 'detail_screen_test.mocks.dart';
+
+Device buildMockDevice(Trait trait) {
+  var mockDevice = Device("TestedDeviceName", "", "", "", "", "",
+      GDateTime('stub'), GDateTime('stub'), [trait]);
+  return mockDevice;
+}
+
+Widget createDetailScreenWhenLoading(
+  Request req,
+  String deviceId,
+) {
+  TraitBasedDeviceNotifier mockTraitBasedNotifier =
+      MockTraitBasedDeviceNotifier();
+  when(mockTraitBasedNotifier.isLoading).thenReturn(true);
+
+  LockProvider mockLockProvider = MockLockProvider();
+
+  return MaterialApp(
+    home: Column(children: [
+      MultiProvider(providers: [
+        ChangeNotifierProvider<TraitBasedDeviceNotifier>.value(
+            value: mockTraitBasedNotifier),
+        ChangeNotifierProvider<LockProvider>.value(value: mockLockProvider),
+      ], child: DetailScreenWidget(req, deviceId)),
+    ]),
+  );
+}
+
+Widget createDetailScreenWidgetForTrait(
+  Trait,
```

Comment:
Not sure how this is working, should this not be passed actual trait object

Reply:
I've pushed a commit to fix this.